presto-main/src/test/java/com/facebook/presto/sql/planner/TestLogicalPlanner.java

```diff
@@ -678,4 +682,19 @@ public void testUsesDistributedJoinIfNaturallyPartitionedOnProbeSymbols()
                                         exchange(REMOTE, REPLICATE,
                                                 node(TableScanNode.class))))));
     }
+
+    @Test
+    public void testDistributedSort()
```

Comment:
would you also update `testJoinWithOrderBySameKey` to confirm plan nodes involved in distributed sort?

Reply:
Would you consider adding a test for the query that motivated this change?

`select *, row_number() over (order by orderkey) from lineitem`

Reply:
> would you also update testJoinWithOrderBySameKey to confirm plan nodes involved in distributed sort?

That test is not really a distributed planner test. Actually I think it's good that the test is agnostic of whether dist sort is enabled or not.

Reply:
> Would you consider adding a test for the query that motivated this change?

`select *, row_number() over (order by orderkey) from lineitem`

Unpartitioned window operator with order by is not yet rewritten on top of distributed ORDER BY. I think that might be a next step, but window operator would need to be extended to support streaming of unpartitioned but ordered windows.